azurerm/internal/services/compute/virtual_machine.go

```diff
@@ -42,3 +52,383 @@ func ValidateVirtualMachineID(i interface{}, k string) (warnings []string, error
 
 	return warnings, errors
 }
+
+func virtualMachineAdditionalCapabilitiesSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				// TODO: confirm this command
+
+				// NOTE: requires registration to use:
+				// $ az feature show --namespace Microsoft.Compute --name UltraSSDWithVMSS
+				// $ az provider register -n Microsoft.Compute
+				"ultra_ssd_enabled": {
+					Type:     schema.TypeBool,
+					Optional: true,
+					Default:  false,
+					ForceNew: true,
+				},
+			},
+		},
+	}
+}
+
+func expandVirtualMachineAdditionalCapabilities(input []interface{}) *compute.AdditionalCapabilities {
+	capabilities := compute.AdditionalCapabilities{}
+
+	if len(input) > 0 {
+		raw := input[0].(map[string]interface{})
+
+		capabilities.UltraSSDEnabled = utils.Bool(raw["ultra_ssd_enabled"].(bool))
+	}
+
+	return &capabilities
+}
+
+func flattenVirtualMachineAdditionalCapabilities(input *compute.AdditionalCapabilities) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	ultraSsdEnabled := false
+
+	if input.UltraSSDEnabled != nil {
+		ultraSsdEnabled = *input.UltraSSDEnabled
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"ultra_ssd_enabled": ultraSsdEnabled,
+		},
+	}
+}
+
+func virtualMachineIdentitySchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"type": {
+					Type:     schema.TypeString,
+					Required: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(compute.ResourceIdentityTypeSystemAssigned),
+						string(compute.ResourceIdentityTypeUserAssigned),
+						string(compute.ResourceIdentityTypeSystemAssignedUserAssigned),
+					}, false),
+				},
+
+				"identity_ids": {
+					Type:     schema.TypeSet,
+					Optional: true,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
```

Comment:
Should we validate this?

Reply:
we should, I'm going to add a TODO and address this in a separate PR since this should be fixed in all the Identity blocks